### ISSUE TITLE
Is it possible to deactive the camo stuff in zulip ?

### ISSUE THREAD

**Issue by thelaoshi** [zulip/docker-zulip#179]
If i try to link images that are not SSL (http://...) they never work, because zulip seems to generate some kinde of https://...../external_content/ url, which ends in a 404.

Seems to be only done with http:// links. https:// just works fine.

Already tried to set the CAMO_URI to '', but this doesnt seem to help aswell.

**Comment by timabbott** [comment]
Thanks for the report @thelaoshi!  @rht and I just noticed yesterday that the `camo` service doesn't seem to be included in this image, and I guess this confirms the failure one would expect from that. 

The fix is that we need to change `dockervoyager.pp` in the main Zulip repo to add `localhost_camo.pp` to the list of modules (or finish the migration to eliminate camo as a dependency, which is mostly merged into master right now).  

That said, setting `CAMO_URI` like that will disable camo, but it only fixes the links in new messages, not ones sent previously.  Can you confirm?

**Comment by rht** [comment]
From git archaeology I found that `camo` used to be included but was subsequently removed by @galexrt.

(There is a relevant named psychology phenomenon where after one sees a new 'x' for the first time, this 'x' suddenly appears more often over time.)

**Comment by thelaoshi** [comment]
Hey @timabbott ,
thanks for the fast reply! 👍 
I use docker-compose and there I added `SETTING_CAMO_URI: ''` which seems to be found by the container, because I see `Empty var for key "CAMO_URI"` in the log.
But after a down/up new `<img src"` elements in the `message_inline_image` div still have a /external_content/.... URI. But I don't now if a camo on/off is maybe saved somewhere else in a volume?

**Comment by thelaoshi** [comment]
Just tested it again with "xxx" as CAMO_URI and that is reflected after a restart. `img src` then is something like `xxx/.....`.
Really looks like setting CAMO_URI to '' doesnt turn of CAMO support. I'm using Zulip 1.9.1 btw.

**Comment by timabbott** [comment]
@thelaoshi can you look at what ends up in `/etc/zulip/settings.py` in the container with that setting?  E.g. `grep CAMO_URI /etc/zulip/settings.py`.

**Comment by thelaoshi** [comment]
@timabbott yeah, it ends up there. Already created a container with standalone camo (additional subdomain) running, to get it to work.

**Comment by timabbott** [comment]
OK.  We're probably doing a major Zulip release in a few weeks, which might include removal of camo from the standard installation process (replacing it with `thumbor`, which already runs in the container).  Assuming that works out soon, that seems like a good approach.

**Comment by thelaoshi** [comment]
sounds very good :-)

**Comment by thelaoshi** [comment]
any news on this one ?

**Comment by timabbott** [comment]
@thelaoshi sorry for the slow reply -- I was ill with the flu when you posted that question.  No news other than this transition wasn't ready in time for the release.  I think there's not a ton more work required to finish it, though.

**Comment by rht** [comment]
> (There is a relevant named psychology phenomenon where after one sees a new 'x' for the first time, this 'x' suddenly appears more often over time.)

https://en.wikipedia.org/wiki/Baader%E2%80%93Meinhof_effect

**Comment by timabbott** [comment]
Because thumbor seems to be EOL upstream, this isn't likely to get a resolution until we can find an alternative replacement for camo.  

So I guess that means we need to add camo to the Docker image.

**Comment by luckyraul** [comment]
Same problem on 4.2

**Comment by jwillmer** [comment]
@luckyraul  Since CAMO is not referenced in [entrypoint.sh](https://github.com/zulip/docker-zulip/blob/master/entrypoint.sh) you need to define a custom variable as described in the readme:

> A new Zulip setting not being passed through the Docker entrypoint.sh script properly. If you run into this sort of problem you can work around it by specifying a ZULIP_CUSTOM_SETTINGS with one setting per line below, but please report an issue so that we can fix this for everyone else.

Something like this should work: `ZULIP_CUSTOM_SETTINGS_CAMO_URI: ''`

But I did not work for me. The setting is not applied. Maybe I'm interpreting the README wrong? @timabbott

**Comment by sahil87** [comment]
Using it this way fixed it: `ZULIP_CUSTOM_SETTINGS: CAMO_URI = ""`

**Comment by timabbott** [comment]
That seems like an OK workaround for now, thanks for figuring it out.  We'll still want to fix this properly by just having camo run in the container like it does for other Zulip servers, since it does provide privacy benefits.
